cmd/crc/cmd/delete.go

```diff
@@ -87,3 +93,49 @@ func (s *deleteResult) prettyPrintTo(writer io.Writer) error {
 	}
 	return nil
 }
+
+// TODO: remove it after we do one more upgrade of the podman bundle from 4.4.1
+func removeCRCHostEntriesFromKnownHosts() error {
+	knownHostsPath := filepath.Join(constants.GetHomeDir(), ".ssh", "known_hosts")
+	f, err := os.Open(knownHostsPath)
+	if err != nil {
+		return fmt.Errorf("Unable to open user's 'known_hosts' file: %w", err)
+	}
+	defer f.Close()
+
+	tempHostsFile, err := os.CreateTemp("", "crc")
+	if err != nil {
+		return fmt.Errorf("Unable to create temp file: %w", err)
+	}
+	defer func() {
+		os.Remove(tempHostsFile.Name())
+		tempHostsFile.Close()
+	}()
+
+	if err := tempHostsFile.Chmod(0600); err != nil {
+		return fmt.Errorf("Error trying to change permissions for temp file: %w", err)
+	}
+
+	var foundCRCEntries bool
+	scanner := bufio.NewScanner(f)
+	writer := bufio.NewWriter(tempHostsFile)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), "[127.0.0.1]:2222") {
+			foundCRCEntries = true
+			continue
+		}
+		_, err := writer.WriteString(fmt.Sprintf("%s\n", scanner.Text()))
+		if err != nil {
+			return fmt.Errorf("Error while writing hostsfile content to temp file: %w", err)
+		}
+	}
+
+	if err := writer.Flush(); err != nil {
+		return fmt.Errorf("Error while flushing buffered content to temp file: %w", err)
+	}
+
+	if foundCRCEntries {
+		return os.Rename(tempHostsFile.Name(), knownHostsPath)
```

Comment:
The only thing is if a user perform `crc delete` and then in different terminal ssh to some server around same time then it might be possible to have a inconsistent state of this file, may we should use some locking mechanism to overcome it?

Reply:
rename will be atomic, i don't think we need a locking mechanism since that other process is outside of CRC i think the os should take care of this automatically?

Reply:
we can also do something like --  keep a backup of the original known_hosts in the same location and if there's an error from `os.Rename` we can restore (rename) the backed-up version